keyboards/telophase/keymaps/grahampheath/keymap.c

```diff
@@ -0,0 +1,293 @@
+// this is the style you want to emulate.
+// This is the canonical layout file for the Quantum project. If you want to add another keyboard,
+
+#include "telophase.h"
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum telophase_layers
+{
+  _QWERTY,
+  _LOWER,
+  _RAISE,
+  _ADJUST,
+  _EMOJI,
+};
+
+enum telophase_keycodes
+{
+  LOWER = SAFE_RANGE,
+  RAISE,
+  ADJUST,
+  EMOJI,
+  ABOVE,
+  BUG,
+  CONFUSED,
+  CRY,
+  FLIP,
+  FNGLEFT,
+  FNGRIGHT,
+  FROWN,
+  GRIN,
+  HEART,
+  JOY,
+  LLAP,
+  SHIT,
+  SHRUG,
+  THMBDN,
+  THMBUP,
+  TOUNGE,
+  WINK
+};
+
+#define LONGPRESS_DELAY 150
+#define LAYER_TOGGLE_DELAY 300
+
+#define KC_X0 MT(MOD_LCTL, KC_ESC)  // Hold for Left Ctrl, Tap for GraveESC
+#define KC_X1 MT(MOD_RSFT, KC_ENT)  // Hold for Right Shift, Tap for Enter
+#define KC_X2 MT(MOD_RSFT, LGUI(KC_ENT))  // Send Command Enter
+#define KC_BACK LGUI(KC_LBRC) // Back
+#define KC_FORWARD LGUI(KC_RBRC) // Forward
+#define KC_EMOJ TT(_EMOJI)  // Hold for Emoji Layer, or tap 5 times.
+#define KC_QS LGUI(KC_SPC)  // Send Command + Space (for QuickSilver).
+#define KC_TABR LGUI(KC_RCBR)  // Send Command + } (for tab changing).
+#define KC_TABL LGUI(KC_LCBR)  // Send Command + { (for tab changing).
+#define HYPR_0 HYPR(KC_TILD)  // Send Hyper + ~.
+#define HYPR_1 HYPR(KC_EXLM)  // Send Hyper + !.
+#define HYPR_2 HYPR(KC_AT)  // Send Hyper + @.
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
+#define XXXXXXX KC_NO
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+ /* Qwerty
+  * ,-----------------------------------------------------------------------------------.
+  * | Esc  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Bksp |
+  * |------+------+------+------+------+-------------+------+------+------+------+------|
+  * | Tab  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  '   |
+  * |------+------+------+------+------+------|------+------+------+------+------+------|
+  * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter |
+  * |------+------+------+------+------+------+------+------+------+------+------+------|
+  * |Emoji | Ctrl | Alt  |Lower | Cmd  |Space |Space |Raise | Left | Down |  Up  |Right |
+  * `-----------------------------------------------------------------------------------'
+  */
+[_QWERTY] = {
```

Comment:
As the default, using the layout version would be better (also, if you use the "layouts" folder/feature, you should only need a single keymap for ALL of the 4x12 boards)